exercises/t_jubb/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,270 @@
+
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <vector>
+
+
+/*
+Shape class
+ */
+class Shape {
+
+public:
+	Shape() { area = 0.0; this->perimeter = 0.0; this->sides = 0; this->type = " "; };
+
+	double get_area() { return area; };
+	double get_perimeter() { return perimeter; };
+	double get_sides() { return sides; };
+	std::string get_type() { return type; };
+	void print_shape() {  
+		int pad = 10 - type.size();
+		std::string pad_blank(pad, ' ');
+		std::cout << "Shape : " << type << pad_blank;
+		std::cout << " Area : " << area << " Perim. : " << perimeter << std::endl;
+		};
+
+protected:
+	double area, perimeter;
+	int sides;
+	std::string type;
+};
+
+/*
+Square class
+*/
+class Square: public Shape {
+
+public:
+	Square() {
```

Comment:
It is generally considered good practice to avoid having a default constructor unless it make sense for that class. In this case for example the default here produces an object in an odd state where it's a square but with zero area and perimeter. It would be better to just remove this constructor and require the height/width to be passed in.